### ISSUE TITLE
False-positive `invalid-enum-extension` with `IntFlag`

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#6295]
### Bug description

```py
from enum import IntFlag

class MyFlags(IntFlag):  # <-- false-positive: invalid-enum-extension
    SUPPORT_OPEN = 1
    SUPPORT_CLOSE = 2
```

The check only tests that the parent doesn't contain an `Assign` node.
https://github.com/PyCQA/pylint/blob/a10ff70f29fa03b465852c98d5a90155730052c5/pylint/checkers/classes/class_checker.py#L842-L848

However, that doesn't work in cases were methods are assigned. Those assignments are valid.
https://github.com/python/cpython/blob/v3.10.4/Lib/enum.py#L996-L998

/CC: @jpy-git Would you mind taking a look at this? I would suggest taking a look at `ancestor.getattr("__members__")` and especially the `dict` which is added by astroid if an enum contains enumeration members.

Ref: #6018

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
E0244: Extending inherited Enum class "IntFlag" (invalid-enum-extension)
```

### Expected behavior

_No error_

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.12.0-dev0
Python 3.10.4 (v3.10.4:9d38120e33, Mar 23 2022, 17:29:05) [Clang 13.0.0 (clang-1300.0.29.30)]
```
